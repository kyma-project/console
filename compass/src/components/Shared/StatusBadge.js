import React from 'react';
import PropTypes from 'prop-types';
import { Badge } from 'fundamental-react/lib/Badge';

class StatusBadge extends React.Component {
  PropTypes = {
    status: PropTypes.string,
  };

  processStatus() {
    const { status } = this.props;
    let type = 'warning';

    switch (status) {
      case 'INITIAL':
        return <Badge>{status}</Badge>;
       case 'READY':
        type = 'success';
        break;
      case 'UNKNOWN':
        type = 'warning';
        break;
      case 'FAILED':
        type = 'error';
        break;
      default:
        type = 'warning';
    }

    return <Badge type={type}>{status}</Badge>;
  }

  render = () => (
    this.processStatus()
  );
}

export default StatusBadge