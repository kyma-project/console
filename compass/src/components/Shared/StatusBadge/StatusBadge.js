import React from 'react';
import { Badge } from 'fundamental-react/Badge';

const StatusBadge = ({ status }) => {
  let type;

  switch (status) {
    case 'INITIAL':
      return <Badge>{status}</Badge>;
    case 'READY':
      type = 'success';
      break;
    case 'UNKNOWN':
      type = 'warning';
      break;
    case 'FAILED':
      type = 'error';
      break;
    default:
      type = 'warning';
  }

  return <Badge type={type}>{status}</Badge>;
};

export default StatusBadge;
