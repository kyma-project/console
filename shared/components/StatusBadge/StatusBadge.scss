.fd-badge.status-badge {
  font-size: var(--sapFontSize, 0.875rem);
  font-family: var(
    --sapFontFamily,
    '72',
    '72full',
    Arial,
    Helvetica,
    sans-serif
  );
  margin: 0;

  max-width: 100%;
  display: inline-block;

  line-height: 1;

  //inverted
  height: auto;
  min-height: 1rem;
  width: auto;
  padding: 0.14rem 0.4rem;
  border-radius: 0.125rem;
  font-weight: 700;
  background-color: var(--sapButton_Neutral_Background, #6a6d70);
  color: var(--sapButton_Neutral_TextColor, #fff);

  &--success {
    background-color: var(--sapButton_Success_Background, #107e3e);
  }
  &--warning {
    background-color: var(--sapButton_Critical_Background, #e9730c);
  }
  &--error {
    background-color: var(--sapButton_Negative_Background, #b00);
  }
  &--info {
    background-color: var(--sapInformativeElementColor, #0a6ed1);
  }

  &.hasTooltip {
    cursor: help;
  }
}
