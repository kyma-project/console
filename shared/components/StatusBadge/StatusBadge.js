import React from 'react';
import { Badge } from 'fundamental-react/Badge';
import PropTypes from 'prop-types';
import './StatusBadge.scss';
import classNames from 'classnames';
import { Tooltip } from '../Tooltip/Tooltip';

const resolveType = status => {
  switch (status.toUpperCase()) {
    case 'INITIAL':
      return 'info';

    case 'READY':
    case 'RUNNING':
    case 'SUCCESS':
      return 'success';

    case 'UNKNOWN':
    case 'WARNING':
      return 'warning';

    case 'FAILED':
    case 'ERROR':
    case 'FAILURE':
      return 'error';

    default:
      return undefined;
  }
};

export const StatusBadge = ({
  status,
  type,
  children,
  autoResolveType = false,
  tooltipProps = {},
}) => {
  if (autoResolveType) type = resolveType(status);

  const classes = classNames('status-badge', {
    ['status-badge--' + type]: type,
    'has-tooltip': children,
  });

  const badgeElement = (
    <Badge role="status" modifier="filled" className={classes}>
      {status}
    </Badge>
  );

  if (children)
    return (
      <Tooltip content={children} {...tooltipProps}>
        {badgeElement}
      </Tooltip>
    );
  else return badgeElement;
};

StatusBadge.propTypes = {
  status: PropTypes.string.isRequired,
  type: PropTypes.oneOf(['success', 'warning', 'error', 'info']),
  autoResolveType: PropTypes.bool,
};
