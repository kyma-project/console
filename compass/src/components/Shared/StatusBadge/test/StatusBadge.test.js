import renderer from 'react-test-renderer';
import StatusBadge from '../StatusBadge';
import React from 'react';

describe('StatusBadge', () => {
  it('should render loading state initially', () => {
    const component = renderer.create(<StatusBadge status="INITIAL" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
