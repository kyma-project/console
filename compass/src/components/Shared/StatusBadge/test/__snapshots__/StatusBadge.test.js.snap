// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBadge should render loading state initially 1`] = `
<span
  className="fd-badge"
>
  INITIAL
</span>
`;
